src/algebraic_geometry/structure_sheaf.lean

```diff
@@ -328,33 +340,265 @@ def stalk_iso_Type (x : prime_spectrum R) :
   ⟨structure_sheaf_stalk_to_fiber_injective R x,
    structure_sheaf_stalk_to_fiber_surjective R x⟩).to_iso
 
--- PROJECT: Improve this to an isomorphism of rings.
+
 /-
-/--
-TODO: this should follow easily from `forget CommRing` preserving filtered colimits.
--/
-def compare_stalks (x : prime_spectrum R) :
-  (forget CommRing).obj ((structure_sheaf R).presheaf.stalk x) ≅ (structure_sheaf_in_Type R).presheaf.stalk x :=
-sorry
 
-/--
-TODO:
+Mathematical Notations:
+
+X = Spec R
+OX = structure sheaf
+
+In the following we construct an isomorphism between OX_p and R_p given any point p corresponding
+to a prime ideal in R.
+
+We do this via 8 steps:
+
+1. def const (f g : R) (V) (hv : V ≤ D_g) : OX(V) [for api]
+2. def to_open (U) : R ⟶ OX(U)
+3. [2] def to_stalk (p : Spec R) : R ⟶ OX_p
+4. [2] def to_basic_open (f : R) : R_f ⟶ OX(D_f)
+5. [3] def localization_to_stalk (p : Spec R) : R_p ⟶ OX_p
+6. def open_to_localization (U) (p) (hp : p ∈ U) : OX(U) ⟶ R_p
+7. [6] def stalk_to_fiber_ring_hom (p : Spec R) : OX_p ⟶ R_p
+8. [5,7] def stalk_iso (p : Spec R) : OX_p ≅ R_p
+
+In the square brackets we list the dependencies of a construction on the previous steps.
+
 -/
-def stalk_to_fiber_ring_hom (x : prime_spectrum R) :
+
+/-- The section of `structure_sheaf R` on an open `U` sending each `x ∈ U` to the element
+`f/g` in the localization of `R` at `x`. -/
+def const (f g : R) (U : opens (Spec.Top R))
+  (hu : ∀ x ∈ U, g ∈ (x : Spec.Top R).as_ideal.prime_compl) :
```

Comment:
I think `\notin (_).as_ideal` is defeq and a bit shorter/more readable?

Reply:
but is harder to use

Reply:
Ok, fair enough